juniper_junos-formula/juniper_junos/files/protocols.j2

```diff
@@ -0,0 +1,63 @@
+{%- set protocols = salt['pillar.get']('juniper_junos:protocols', {}) -%}
+{%- set protocol_names = protocols.keys() -%}
+
+{%- set setprotocol = 'set protocols ' %}
+
+{%- for protocol, pconfig in protocols.items() %}
+{%- set setprotocol = setprotocol ~ protocol %}
```

Comment:
If there are multiple protocols, you'll end up with (assuming "foo" and "bar"):

    set protocols foobar

when the loop runs the second time.

To fix this, you can move

    {%- set setprotocol = 'set protocols ' %}

inside the for loop so that it gets reset.

Reply:
Hi, good point, thanks for spotting!